Added Testimonial Page and Form

Nishitbaria opened the pull request:
# Related Issues #434 
-I have designed a testimonial section  and created a testimonial form for our website. This pull request adds the necessary HTML, CSS, and JavaScript code to implement the testimonial functionality.
  
## Changes Made
- Created a new testimonial section with HTML and CSS,Bootstrap.
- Added JavaScript functionality to dynamically display testimonials in cards.
- Implemented a testimonial form that captures user input.
- Created an API endpoint on the server to receive and store testimonial data.
- Updated the necessary routes and handlers to handle testimonial requests.
  
# Additional Information
- Any additional information or context
  
# Checklist
- [x] Tests
- [x] Translations
- [x] Documentations

# Screenshots

![image](https://github.com/Opentek-Org/opentek/assets/85815172/4c934776-6611-4112-bbd0-5e173760659e)

See the **Video**
https://github.com/Opentek-Org/opentek/assets/85815172/51c35615-852b-46dd-b1f2-349cc2d8f10d
@AnirudhPanda ,please review and merge

Nishitbaria commented:
@AnirudhPanda ,please review and merge my pull request

AnirudhPanda commented on server/app.js:
```diff
@@ -0,0 +1,139 @@
+  }
+});
+
+app.get("/testimonials", async (req, res) => {
+  try {
+    const testimonials = await Testimonial.find();
+    res.status(201).json({ testimonials });
+  } catch (error) {
+    console.error("Error retrieving testimonials:", error);
+    res.status(500).json({ error: "Error retrieving testimonials" });
+  }
+});
```
@Nishitbaria

AnirudhPanda commented:
also, is the add testimonial fully functional?

AnirudhPanda commented:
@sunny0625 check this PR

Nishitbaria commented:
> also, is the add testimonial fully functional?

yes it is full functional see video ,i have shown demo

AnirudhPanda commented:
the add button one?

On Tue, Jul 11, 2023 at 4:28 PM Nishitbariya ***@***.***>
wrote:

> also, is the add testimonial fully functional?
>
> yes it is full functional see video ,i have shown demo
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Opentek-Org/opentek/pull/444#issuecomment-1630611921>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APZGUAAOU7ME3WUOP4L4D3DXPUWTRANCNFSM6AAAAAA2FAETFQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Thanks and Regards,
*Anirudh Panda*

Nishitbaria commented:
> the add button one?
> […](#)
> On Tue, Jul 11, 2023 at 4:28 PM Nishitbariya ***@***.***> wrote: also, is the add testimonial fully functional? yes it is full functional see video ,i have shown demo — Reply to this email directly, view it on GitHub <[#444 (comment)](https://github.com/Opentek-Org/opentek/pull/444#issuecomment-1630611921)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/APZGUAAOU7ME3WUOP4L4D3DXPUWTRANCNFSM6AAAAAA2FAETFQ> . You are receiving this because you were mentioned.Message ID: ***@***.***>
> -- Thanks and Regards, *Anirudh Panda*

yes adding testimonial functionality is working ,in video i have shown by adding testimonial

AnirudhPanda commented:
it does not seem to work!!
make a detailed video please

On Tue, Jul 11, 2023 at 4:32 PM Nishitbariya ***@***.***>
wrote:

> the add button one?
> … <#m_2307669561207328261_>
> On Tue, Jul 11, 2023 at 4:28 PM Nishitbariya *@*.*> wrote: also, is the
> add testimonial fully functional? yes it is full functional see video ,i
> have shown demo — Reply to this email directly, view it on GitHub <#444
> (comment)
> <https://github.com/Opentek-Org/opentek/pull/444#issuecomment-1630611921>>,
> or unsubscribe
> https://github.com/notifications/unsubscribe-auth/APZGUAAOU7ME3WUOP4L4D3DXPUWTRANCNFSM6AAAAAA2FAETFQ
> <https://github.com/notifications/unsubscribe-auth/APZGUAAOU7ME3WUOP4L4D3DXPUWTRANCNFSM6AAAAAA2FAETFQ>
> . You are receiving this because you were mentioned.Message ID: @.*>
> -- Thanks and Regards, *Anirudh Panda*
>
> yes adding testimonial functionality is working ,in video i have shown by
> adding testimonial
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Opentek-Org/opentek/pull/444#issuecomment-1630618626>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APZGUAEJEF5FTBLR67LM7MDXPUXF5ANCNFSM6AAAAAA2FAETFQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 

Thanks and Regards,
*Anirudh Panda*

Nishitbaria commented:
@AnirudhPanda ,You have to refresh and you can see your cards:
![image](https://github.com/Opentek-Org/opentek/assets/85815172/6d2f0675-b563-4502-814c-71a1068931b6)

Nishitbaria commented:
check this 

https://github.com/Opentek-Org/opentek/assets/85815172/c0ccda42-bcd1-427c-9f7f-9135dcfdb21a

Nishitbaria commented:
> @AnirudhPanda ,     check this
> 
>  PassFab20230711_170025262.mp4

Nishitbaria commented on server/app.js:
```diff
@@ -0,0 +1,156 @@
+    const testimonialData = req.body;
+
+    // Check if testimonial with the same name already exists
+    const existingTestimonial = await Testimonial.findOne({
+      name: testimonialData.name,
+    });
```
Its not a problem

Nishitbaria commented:
@AnirudhPanda ,please review and merge

AnirudhPanda reviewed (changes requested):
few changes to be made:

- after we add testimonial there should be a success response (this maybe a hard refresh too)
- carousel of cards should be made manual movement too instead of just waiting for thee cards
- I do not see any rate limiting warning while putting numbers more than 4

Nishitbaria commented:
I have updated Validation for URL and Rating ( it should between 1 and 5) and also added respected Alerts and also page refreshes after successfully Added 

![image](https://github.com/Opentek-Org/opentek/assets/85815172/7316148d-00c8-48a4-8003-8f411f438926)
![image](https://github.com/Opentek-Org/opentek/assets/85815172/7f0bd2cd-07a9-40f6-8ba8-13c7b245e638)
![image](https://github.com/Opentek-Org/opentek/assets/85815172/d2fde680-655b-49f8-955d-a8d2c443fe7a)

AnirudhPanda commented:
update me once it's all done man!!
great going @Nishitbaria

Nishitbaria commented:
@AnirudhPanda ,I have updated ,please review and merge pull request

AnirudhPanda commented:
![image](https://github.com/Opentek-Org/opentek/assets/66218496/d479d750-5cb1-4570-84ce-f15365190814)
the URL is a valid URL - https://avatars.githubusercontent.com/u/66218496?v=4

Nishitbaria commented:
> https://avatars.githubusercontent.com/u/66218496?v=4

The previous URL validation was not for link with params like `?v=4`

So i have updated now

AnirudhPanda reviewed (commented):
Okay, so when we add a testimonial it gets written in paste.js? So if we want to remove the testimonial we can remove it?
Like SPAM etc?
or else can we do an email validation too to ensure only one testimonial per person? Maybe in another PR

Nishitbaria commented:
> Okay, so when we add a testimonial it gets written in paste.js? So if we want to remove the testimonial we can remove it? Like SPAM etc? or else can we do an email validation too to ensure only one testimonial per person? Maybe in another PR

@AnirudhPanda  It is saved in mongo db database ,Yes i will create issue for email validation to remove spam  ,please merge this and I will start working on that

AnirudhPanda commented:
> It is saved in mongo db database

how can I use it to remove a testimonial? Any collab access there?

Nishitbaria commented:
>@AnirudhPanda ,In upcoming issues i will make route to delete testimonial by that functionality you can delete and if you want access to database,i will tell you how to create MongoDB database andhave access to it using MongoDB atlas

AnirudhPanda commented:
okay cool. SO merging this PR just clear all the testimonials we have currently

Nishitbaria commented:
> okay cool. SO merging this PR just clear all the testimonials we have currently

@AnirudhPanda  Done

Nishitbaria commented:
> > okay cool. SO merging this PR just clear all the testimonials we have currently
> 
> @AnirudhPanda Done

Hey @AnirudhPanda   I will Creating new Pr in 5min please also Review it :)

AnirudhPanda commented:
> > > okay cool. SO merging this PR just clear all the testimonials we have currently
> > 
> > 
> > @AnirudhPanda Done
> 
> Hey @AnirudhPanda I will Creating new Pr in 5min please also Review it :)

sure
